src/Metadata.tsx

```diff
@@ -0,0 +1,127 @@
+import { KalturaClient } from "kaltura-typescript-client";
+import {
+  KalturaMediaEntry,
+  KalturaUploadToken,
+} from "kaltura-typescript-client/api/types";
+import { KalturaMediaType } from "kaltura-typescript-client/api/types/KalturaMediaType";
+import { BasicEntryMetadata, createEntryForUpload } from "KalturaModule";
+import { set } from "lodash";
+import * as React from "react";
+import { ChangeEvent, useState } from "react";
+import { useErrorHandler } from "react-error-boundary";
+
+export interface MetadataProps {
+  /**
+   * A prefix to add to the DOM `id` for the root of this component.
+   */
+  idPrefix: string;
+  /**
+   * A `KaltruaClient` that has been setup with valid session token and partner ID, possibly using
+   * `createClient` from `KalturaModule`.
+   */
+  kClient: KalturaClient;
+  /**
+   * The media type for the Media Entry which is to be created - typically matching the type of file
+   * uploaded which resulted in the value provided for `uploadResult`.
+   */
+  mediaType: KalturaMediaType;
+  /**
+   * Callback triggered on the successful creation of the Media Entry.
+   *
+   * @param entry Details returned from the server for the new Media Entry.
+   */
+  onEntryCreated: (entry: KalturaMediaEntry) => void;
+  /**
+   * The result of a previously successful upload - such as one completed with the `Upload`
+   * component.
+   */
+  uploadResult: KalturaUploadToken;
+}
+
+/**
+ * Having already uploaded a media file to Kaltura, this component then allows for the configuration
+ * of some basic metadata and the creation of an enclosing Media Entry on Kaltura.
+ */
+export const Metadata = ({
+  idPrefix,
+  kClient,
+  mediaType,
+  onEntryCreated,
+  uploadResult,
+}: MetadataProps): JSX.Element => {
+  // Alias type to bind the below two expressions to the same type
+  type FormTarget = BasicEntryMetadata;
+  const [form, setForm] = useState<FormTarget>({
+    name: "",
+    description: "",
+    mediaType,
+  });
+  // Basic compile time validator for form fields
+  const formField = (name: keyof FormTarget) => name;
+
+  const [error, setError] = useState<string>("");
+  const handleError = useErrorHandler();
```

Comment:
I am not very clear about  why use both `handleError` and `setError` .  When should we call `setError` and when to call `handleError` ?

Reply:
Yeah, I can see how that makes little sense to others. Oops.

The intention here is:

- `setError` would be use for messages to display to the user so they can try and recover from the situation - and would also support any form validation we might need (but I'm no longer sure we'll need any validation); and
- `handleError` is to explicitly handle exceptional situations - .e.g Exceptions - where we simply want to make sure we trigger the Error Boundary and don't simply have a blank component due to uncaught exceptions.

I'll add some more explicit naming around this methinks - like `setErrorMessage` and `triggerErrorBoundary`. :thinking: